pyclowder/extractors.py

```diff
@@ -165,9 +173,18 @@ def start(self):
                                               rabbitmq_queue=self.args.rabbitmq_queuename,
                                               mounted_paths=json.loads(self.args.mounted_paths),
                                               clowder_url=self.args.clowder_url,
-                                              max_retry=self.args.max_retry)
+                                              max_retry=self.args.max_retry,
+                                              extractor_key=self.args.extractor_key,
+                                              clowder_email=self.args.clowder_email)
                 connector.connect()
                 connector.register_extractor(self.args.registration_endpoints)
+
+                # TODO: register extractor initially without _process_message?
```

Comment:
how is this different from the `register_extractor` also I like to move away from this in the future.

Reply:
register_extractor registers with arg-specified endpoints, while this is a generic registration with the clowder instance in clowder_url (if not already registered there). this would allow extractor_info to not have registration_urls, but auto-register with Clowder. i was having trouble registering with just RMQ heartbeats in my test environment, but maybe this is unnecessary...